File: androidApp/src/main/kotlin/com/crossfolio/android/ui/TabBarScreen.kt
package com.crossfolio.android.ui

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.NavigationBar
import androidx.compose.material3.NavigationBarItem
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.crossfolio.shared.navigation.AppTab
import com.crossfolio.shared.navigation.TabBarCoordinator

@Composable
fun TabBarScreen(
    coordinator: TabBarCoordinator = remember { TabBarCoordinator() },
) {
    val state by coordinator.state.collectAsState()

    MaterialTheme {
        Scaffold(
            bottomBar = {
                NavigationBar {
                    AppTab.entries.forEach { tab ->
                        NavigationBarItem(
                            selected = state.selectedTab == tab,
                            onClick = { coordinator.selectTab(tab) },
                            icon = {},
                            label = { Text(tab.title) },
                        )
                    }
                }
            },
        ) { contentPadding ->
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(contentPadding),
            ) {
                when (state.selectedTab) {
                    AppTab.PORTFOLIO -> PortfolioScreen(coordinator.portfolioCoordinator)
                    AppTab.ANALYTICS -> AnalyticsScreen(coordinator.analyticsViewModel)
                    AppTab.PROFILE -> ProfileScreen(coordinator.profileViewModel)
                }
            }
        }
    }
}

private val AppTab.title: String
    get() = when (this) {
        AppTab.PORTFOLIO -> "Портфолио"
        AppTab.ANALYTICS -> "Аналитика"
        AppTab.PROFILE -> "Профиль"
    }

@Preview(showBackground = true)
@Composable
private fun TabBarScreenPreview() {
    TabBarScreen()
}

File: appleApp/Shared/TabBarScreen.swift
import Shared
import SwiftUI

struct TabBarScreen: View {
    private let coordinator: TabBarCoordinator
    @State private var selectedTab: AppTab

    init(coordinator: TabBarCoordinator = TabBarCoordinator(
        portfolioCoordinator: PortfolioCoordinator(),
        analyticsViewModel: AnalyticsViewModel(),
        profileViewModel: ProfileViewModel()
    )) {
        self.coordinator = coordinator
        _selectedTab = State(
            initialValue: (coordinator.state.value as? TabBarState)?.selectedTab ?? .portfolio
        )
    }

    var body: some View {
        TabView(selection: selection) {
            PortfolioScreen(coordinator: coordinator.portfolioCoordinator)
                .tabItem { Label("Портфолио", systemImage: "briefcase") }
                .tag(AppTab.portfolio)

            AnalyticsScreen(viewModel: coordinator.analyticsViewModel)
                .tabItem { Label("Аналитика", systemImage: "chart.bar") }
                .tag(AppTab.analytics)

            ProfileScreen(viewModel: coordinator.profileViewModel)
                .tabItem { Label("Профиль", systemImage: "person") }
                .tag(AppTab.profile)
        }
    }

    private var selection: Binding<AppTab> {
        Binding(
            get: { selectedTab },
            set: { tab in
                coordinator.selectTab(tab: tab)
                selectedTab = tab
            }
        )
    }
}

#Preview {
    TabBarScreen()
}
